assets/javascripts/discourse/components/event-date.gjs

```diff
@@ -0,0 +1,93 @@
+import Component from "@glimmer/component";
+import { inject as service } from "@ember/service";
+import I18n from "discourse-i18n";
+import guessDateFormat from "../lib/guess-best-date-format";
+
+export default class EventDate extends Component {
+    @service siteSettings;
+
+    <template>
+    {{#if this.shouldRender}}
+        <div class="header-topic-title-suffix-outlet event-date-container">
+            {{#if this.siteSettings.use_local_event_date}}
+                <span class="event-date event-local-date past" data-starts_at={{this.eventStartedAt}} title={{this.dateRange}} data-ends_at={{this.eventEndedAt}}>
+                    {{this.localDateContent}}
+                </span>
+            {{else}}
+                <span class="event-date event-relative-date {{this.relativeDateType}}" title={{this.dateRange}} data-starts_at={{this.eventStartedAt}} data-ends_at={{this.eventEndedAt}}>
+                    {{#if this.isWithinDateRange}}
+                        <div class="indicator"></div>
+                        <span class="text">{{this.timeRemainingContent}}</span>
+                    {{else}}
+                        {{this.relativeDateContent}}
+                    {{/if}}
+                </span>
+            {{/if}}
+        </div>
+    {{/if}}
+    </template>
+
+    get shouldRender() {
```

Comment:
Indentation is off on all the JS parts as well ;) I know this is nitpicky, but we've gotta be consistent.